### ISSUE TITLE
Enter the payment type for non-online payments

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#2934]
* As an IAO analyst
* I want to be required to enter the type of payment when a fee has been paid off line
* so that I have a record of how the applicant paid the fee

**Assumptions & Scope**	
What are the assumptions for this story?
Intake needs to enter the payment type when they manually enter a payment received offline (i.e., not by PayBC).

This could include an applicant dialling in by phone to pay by credit card.

Payment types are: Credit card, cheque, money order, and cash

Payment type can be edited (i.e., changed provided the form is approved and a manual entry has been paid)

While rare, applicant may use a different method for deposit vs the balance. If no deposit, they should use the balance field.

What is IN scope?
A drop down with mandatory selection of how an applicant paid the request.

What is NOT in scope?
IAO can already do a manual entry when an applicant has paid.
**Acceptance Criteria**	

**Scenario 1: Payment Type dropdown on unapproved form**
* GIVEN a. user is an IAO analyst  
* WHEN the analyst is on an unapproved CFR Form
* THEN the Payment Type dropdowns will be uneditable and greyed out
* AND the dropdown will be titled "Payment Type"

**Scenario 2: No Payment Type dropdown for Ministry Coordinator**
* GIVEN a user is a ministry coordinator 
* WHEN the coordinator is on the CFR Form
* THEN no Payment Type dropdown will appear

**Scenario 3:  Payment Type dropdown on an approved form**
* GIVEN a user is an IAO analyst
* WHEN the analyst is on an Approved CFR Form
* AND no entry has been paid in the Paid Amount field
* THEN there will be Estimate Payment Type dropdown 
* AND a Balance Payment Type dropdown for full
* AND the dropdown will be uneditable 
* AND the dropdown will be titled "Estimate Payment Type" and "Balance Payment Type"

**Scenario 4: Editable Payment Type**
* GIVEN a user is an IAO analyst on an Approved CFR Form
* WHEN I manually enter an amount in the amount paid field
* THEN Payment Type dropdowns will be editable
* AND the Save button will be disabled until a Payment Type is selected in one of the dropdowns

**Scenario 5: Payment Type Options**
* GIVEN an IAO analyst has manually entered an amount in the amount paid field
* AND the form is Approved
* WHEN the analyst clicks on a Payment Type Dropdown
* THEN the drop down will open
* AND the choices will be: Credit card - Phone, Cheque, Money order, Cash

**Scenario 6: Payment Type Options**
* GIVEN an IAO analyst has manually entered an amount in the amount paid field
* AND the form is Approved
* WHEN the analyst clicks on a Payment Type Dropdown
* THEN the drop down will open
* AND the choices will be: Credit Card Over Phone, Cheque, Money Order, Cash

**Scenario 7: Select Payment Type**
* GIVEN an IAO analyst has opened a Payment Type dropdown
* WHEN the analyst clicks on a Payment Type (e.g., Cheque)
* THEN the drop down will close
* AND the payment type will be selected (e.g., Cheque)
* AND the payment type will be displayed (e.g., Cheque)

**Scenario 8: PayBC Payment Type**
* GIVEN an an applicant has paid a fee in full or as a deposit on PayBC
* WHEN an IAO Analyst is on the CFR Form
* THEN the Payment Type drop down will be uneditable
* AND the payment type will be selected (e.g., Credit Card Online)
* AND the payment type will be displayed (e.g., Credit Card Online)

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@m-prodan, @arielleandrews, could you please review the AC above and make any changes required.

@arielleandrews, let us know if this is required for the Payment Release on Nov 10th or can come later?
 
cc @JHarrietha-AOT, @abin-aot

**Comment by arielleandrews** [comment]
Thanks @lmullane. My preference would be to have this included in the first release. Tracking payments is quite a manual process right now and if we can drive the desired behaviour from the outset that will really help with the accounting responsibilities.

**Comment by lmullane** [comment]
Thanks, @JHarrietha-AOT. Looks good. 

Once change. Could we differentiate between online credit card payment and over the phone credit card payment?

**Comment by JHarrietha-AOT** [comment]
@lmullane Dropdown is updated with additional options and Pay Online is greyed out - as it would not be available for user to select only auto-populated

**Comment by JHarrietha-AOT** [comment]
@lmullane I have divided the dropdown in two based off of Arielle's recommendations.

https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/912622803

She will let me know tomorrow if Money Order/Cheque should be combined into one options or separated into two. Will comment here when I hear back

**Comment by lmullane** [comment]
Great. Looks good.

Thanks @arielleandrews and @JHarrietha-AOT

**Comment by nkan-aot** [comment]
clarification to scenario 4 during standup - if amount paid was changed from $0, only estimate payment method will be mandatory. If amount paid was already greater than $0 and more was added, then balance payment method will be mandatory

**Comment by liseandtea** [comment]
Passed with the following notes:
Scenario 3: This function works only after the request has been put on hold. If the request is still in the "Fee Estimate" stage, the payment type options remain greyed out even though the form is "approved". 
Scenario 6: Is a duplicate of scenario 5.
